Highlight Community Aspect

willirath opened the pull request:
This tries to highlight the community aspect of Pangeo by re-focusing the description of Pangeo.  It also tries to list a few specific achievements that were triggered / facilitated by the collaborative approach within Pangeo.

willirath committed:
Re-focus abstract to the community aspect

willirath committed:
Focus community section

willirath committed:
Highlight the impact of Pangeo on the tech stack

willirath committed:
Add citations for Dask, xarray, and Jupyter

willirath committed:
Shorten tech stack description

willirath committed:
Streamline Applications section

willirath committed:
Streamline ongoing section

willirath committed:
Add PDF and bbl file

willirath commented:
@guillaumeeb My remarks for highlighting the _Pangeo is a community_ aspect.

guillaumeeb commented on pangeo-cnes.tex:
```diff
@@ -54,12 +54,27 @@
-In this paper, we first describe Pangeo: its motivation, community, the underlying technology stack and associated deployments, different applications and the ongoing work. We then present its use at CNES: the context, HPC deployment, and several use cases. We will then conclude with a future outlook for Pangeo in this agency.
+Pangeo\cite{b1} is a community driven effort for open-source big data initially
+focused on the Earth System Sciences.  Its primary goal is to enable scientists
+in analyzing peta-scale datasets both on classical high-performacnce computing
```
Typo in high performance. And we should probably keep something like "one of its primary goal", the community may not agree on the fact that it's THE primary goal.

willirath replied:
I agree. "one of its primary goals" is much better.

guillaumeeb commented on pangeo-cnes.tex:
```diff
@@ -83,75 +98,89 @@ \subsection{Motivations, mission and goals}
 \end{itemize}
 
-Such applications are most effective when deployed on cloud-optimized datasets. This means that data must be accessible in a cloud or distributed storage friendly format, like Zarr for multi-dimensional data, Cloud optimized Geotiff for satellite imagery, or Parquet for tabular data. See \cite{b7} for more details on this crucial point. Formats like NetCDF or CSV files are not adapted or optimize ...
+Applications combining large data sets from different sources are most effective when deployed on cloud infrastructure.  This means that data must be accessible in a cloud or distributed storage friendly format, like Zarr for multi-dimensional data, Cloud optimized Geotiff for satellite imagery, or Parquet for tabular data. See \cite{b7} for more details on this crucial point.
```
I do not agree wit the cloud infrastructure. The key for me is really to have distributed processing friendly file format, that can be use both on Cloud or HPC or other infrastructure.

willirath replied:
I've tried re-formulating this.

guillaumeeb commented on pangeo-cnes.tex:
```diff
@@ -83,75 +98,89 @@ \subsection{Motivations, mission and goals}
-\item It has recently been decided to offer a specific cloud environment for the different scientific communities, in order to cope with each domain specific software stack.
-\item In order to achieve the above point, intensive work is currently being done on simplifying Cloud environment updates, by using modern tooling for Continous Deployment using Hubploy and CircleCI.
-\item As Pangeo is compatible not only with cloud but with different infrastructure, several Dask subcomponents share some logic for deploying clusters. A work to rationalize this logic and separate resources management from Dask scheduling is also ongoing.
+Advancing both, the community and the technological stack, is currently done along the following lines:
```
I'm not an english expert, but this sentence kind of sounds wrong to me.

willirath replied:
Changed.

guillaumeeb commented on pangeo-cnes.tex:
```diff
@@ -83,75 +98,89 @@ \subsection{Motivations, mission and goals}
-\item As Pangeo is compatible not only with cloud but with different infrastructure, several Dask subcomponents share some logic for deploying clusters. A work to rationalize this logic and separate resources management from Dask scheduling is also ongoing.
+Advancing both, the community and the technological stack, is currently done along the following lines:
+\begin{itemize}
+\item To facilitate work with the Pangeo software stack at scale, there is a Binder deployment\cite{b8} providing access to a cloud-based platform with a single click.
```
I would not say to facilitate work.  "To facilitate and learn how to use Pange at scale" ?

willirath replied:
Changed.

guillaumeeb commented on pangeo-cnes.tex:
```diff
@@ -83,75 +98,89 @@ \subsection{Motivations, mission and goals}
+\begin{itemize}
+\item To facilitate work with the Pangeo software stack at scale, there is a Binder deployment\cite{b8} providing access to a cloud-based platform with a single click.
+\item To actively foster collaboration and to make sure that Pangeo remains a diverse and inclusive project, an emerging focus is on community governance.
+\item To better cope with domain-specific software and hardware requirements at this stage of the development, the formerly single example deployment of Pangeo has been split into many specific environments.  This helps with reaching out to new fields and increasingly simplifies automated cloud deployments.
```
I'd like to keep something on CI/CD of Pangeo environment in the cloud.

willirath replied:
Added again.

guillaumeeb commented on pangeo-cnes.tex:
```diff
@@ -83,75 +98,89 @@ \subsection{Motivations, mission and goals}
+\item To facilitate work with the Pangeo software stack at scale, there is a Binder deployment\cite{b8} providing access to a cloud-based platform with a single click.
+\item To actively foster collaboration and to make sure that Pangeo remains a diverse and inclusive project, an emerging focus is on community governance.
+\item To better cope with domain-specific software and hardware requirements at this stage of the development, the formerly single example deployment of Pangeo has been split into many specific environments.  This helps with reaching out to new fields and increasingly simplifies automated cloud deployments.
+\item To reach out to possible new collaborators but also to enable more and more interested scientists, Pangeo members offer workshops in various settings.
```
Maybe add something on interoperability here. Saying that both HPC or Cloud guyes are working together to provide smooth integration and interoperability between various environment?

willirath replied:
I'm not sure this fits here.  In the Applications ssec, there's already a statement on the collaboration of HPC and cloud users / developers.  My gut feeling is that any highlighting of this aspect should also go there.

guillaumeeb commented on pangeo-cnes.bbl:
```diff
@@ -0,0 +1,33 @@
+  Press, 2016.
+\newblock \doi{10.3233/978-1-61499-649-1-87}.
+
+\end{thebibliography}
```
Why is there a separate bibliography here? I'm not a LateX expert...

willirath replied:
Fixed.  This was meant to go away before I remove the [WIP] tag.  But note that the refs are not ordered in a very good way now.  (But this is sth that is best done when the text is finished.)

guillaumeeb reviewed (commented):
Many thanks for the review and proposed changes that generally looks really good to me.

I left some comments, but some of them may be addressed later on if you want.

guillaumeeb commented:
Oh, and please add yourself to the Authors!

willirath committed:
Account for review

This follows https://github.com/guillaumeeb/bids2019/pull/5#pullrequestreview-186277617

willirath commented:
Additional comments:

- I'd drop Figure 3 and just describe it.  I agree that these dash boards are very impressive.  But it's very difficult to read this figure in the PDF.

- I'd transform Figure 4 to a code listing.

apatlpo commented on pangeo-cnes.tex:
```diff
@@ -83,75 +99,94 @@ \subsection{Motivations, mission and goals}
-\item Dask is a library for parallel computing that coordinates with Python’s existing scientific software ecosystem, including libraries cited above. In many cases, it offers users the ability to take existing workflows and quickly scale them to much larger applications. Dask-distributed is an extension of Dask that facilitates parallel execution across many computers.
-\item Xarray is the interface for working with big datasets: it offers a Pandas like API for dealing with labelled n-dimensionnal arrays with a self-describing scientific data capability and strong backends to standard and upcoming community data file formats and access (e.g. netCDF, GeoTIFFs, OPeNDAP).
+\item Dask is a library for parallel and distributed computing that coordinates with Python’s existing scientific software ecosystem.  In many cases, it offers users the ability to take existing workflows and quickly scale them to much larger applications.
+\item Xarray is the interface for working with big datasets: it provides a Pandas-like API for labelled n-dimensional arrays and has backends for established and upcoming self-describing community data file formats and access protocols like netCDF, GeoTIFF, OPeNDAP, and Zarr.  Xarray transparently integrates Dask arrays and hence enables users to easily scale their work to massively parallel comp ...
```
Do we want references for netCDF, GeoTIFF, OPeNDAP and Zarr?

... enables users to ~~easily~~ scale their work ...

guillaumeeb replied:
I think it's important to name the standard Xarray complies with.

And for _easily_, this paper is also for promoting Pangeo ecosystem, but I'm happy to remove it.

apatlpo commented on pangeo-cnes.tex:
```diff
@@ -83,75 +99,94 @@ \subsection{Motivations, mission and goals}
+\subsection{Deployments}
+\label{ssec:deployments}
+
+The GitHub community offers online documentation, scripts and other tools to link them together in order to deploy a Pangeo platform (see \ref{pangeo_stack}) and put the software stack on HPC systems or in the public cloud. The main elements allowing to build and use the platforms along the core libraries are first a set of scripts and documentation that allows automatically creating the necessar ...
```
Does the `Deployments` subsection needs rewriting?
I have the feeling it could be clearer and sharper.

guillaumeeb replied:
If you can propose something, I'm all hears :)! I will try to merge this soon, feel free to open another PR!

apatlpo commented on pangeo-cnes.tex:
```diff
@@ -83,75 +99,94 @@ \subsection{Motivations, mission and goals}
 \label{ssec:applications}
 
-Pangeo's first scientific domain is earth sciences: particularly the domains of oceanographic, terrestrial, atmospheric, and climate sciences. There are already many applications documented in these fields. Some real science use cases can be found online\cite{b5}; these examples, along with their data, can be directly executed from a web browser. We can for example cite \textit{Sea Surface Altime ...
+Pangeo's first scientific domain are the earth-system sciences. There are already many applications documented in these fields. Some real science use cases can be found online\cite{b5}; these examples, along with their data, can be directly executed on cloud infrastructure and be controlled from a web browser.
```
these examples ~~, along with their data,~~ can be directly executed on cloud infrastructure ~~and be controlled~~ from a web browser.

data is not executed

guillaumeeb replied:
Sounds fair

apatlpo commented on pangeo-cnes.tex:
```diff
@@ -83,75 +99,94 @@ \subsection{Motivations, mission and goals}
+The community and the technological stack are currently advanced along the
+following lines:
+\begin{itemize}
+\item To facilitate learning and using the Pangeo software stack at scale, there is a Binder deployment\cite{b8} providing access to a cloud-based platform with a single click.
```
To facilitate learning and using the Pangeo software stack at scale, a Binder deployment\cite{b8} aims at providing access to a cloud-based platform with a single click.

guillaumeeb replied:
I think we should try to explain this better as this was one major comment.

apatlpo reviewed (commented):
minor comments pointing at spots where clarity could be improved

guillaumeeb commented:
Okay, I'm merging.

@apatlpo it would be good if you could provide a PR with your remarks.

Also, I would really like talking about your use case https://github.com/apatlpo/mit_equinox/blob/halrefactor/hal/rechunk_rotspectra.ipynb, but I didn't find the time to dive into it yet. Could you try describing it?

apatlpo commented:
sorry for the slow time response
I'm not sure my comments are worth a PR.
I don't feel I'm qualified enough to clarify "Deployments" section unfortunately.

About the rechunk_rotspectra use case, what is it you'd like me to do precisely? I'm not sure I understand

guillaumeeb commented:
@apatlpo that would be really nice if you could describe in a few lines what you're doing in your use case:
- What is the goal or purpose of the science your doing,
- What are the main steps of the computation,
- What Pangeo enables you to do that was not possible or difficult before.

apatlpo commented:
sure I can do that, when would you need it?

guillaumeeb commented:
For tuesday 😅
